e2e-tests/src/test/electing_validators.rs

```diff
@@ -35,16 +36,25 @@ fn assert_validators_are_elected_stakers<C: AnyConnection>(
     );
 }
 
+// There are v non-reserved validators and s non-reserved seats. We will have seen all
+// the non-reserved validators after ceil(v / s).
+fn min_num_sessions_to_see_all_non_reserved_validators(
+    non_reserved_count: u32,
+    non_reserved_seats: u32,
+) -> u32 {
+    (non_reserved_count as f64 / non_reserved_seats as f64).ceil() as u32
```

Comment:
nit: I know that for non_reserved_seats equal to 0 this will correctly output 0 but still I would handle 0 'specially' here

Reply:
Modified.